src/main/java/stashpullrequestbuilder/stashpullrequestbuilder/StashRepository.java

```diff
@@ -196,6 +196,8 @@ private boolean hasCauseFromTheSamePullRequest(
         if (cause instanceof StashCause) {
           StashCause sc = (StashCause) cause;
           if (StringUtils.equals(sc.getPullRequestId(), pullRequestCause.getPullRequestId())
+              && StringUtils.equals(
```

Comment:
it would be nice if we could get some unit tests for the build matching
probbably it would make sense to add them in a separate class as StashRepositoryTest is quite big already

Reply:
I'll try to add tests. I think it's OK to have many test as long as they are well organized within the file.

I hope to split a class for individual pull requests from `StashRepository` (`StashPullRequest` perhaps), which should split the unit tests as well. But that would be a big change that would conflict with most PRs, so it's better to get small things merged first.

Reply:
My point is: if you can separate the test cases into several logical sets why not do it? 
Of course if you find this situation it suggests the SUT could be refactored to match it, but splitting the tests should help in doing so.